internal/handler/v2/config/dataagreement/config_list_dataattributes_for_dataagreement.go

```diff
@@ -11,14 +11,49 @@ import (
 	"github.com/bb-consent/api/internal/dataagreement"
 	"github.com/bb-consent/api/internal/paginate"
 	"github.com/gorilla/mux"
+	"go.mongodb.org/mongo-driver/bson/primitive"
 )
 
+type dataAgreementForDataAttribute struct {
+	Id      string `json:"id" bson:"_id,omitempty"`
+	Purpose string `json:"purpose"`
+}
+
+type dataAttributeForLists struct {
+	Id            primitive.ObjectID            `json:"id" bson:"_id,omitempty"`
+	Name          string                        `json:"name" valid:"required"`
+	Description   string                        `json:"description" valid:"required"`
+	Sensitivity   bool                          `json:"sensitivity"`
+	Category      string                        `json:"category"`
+	DataAgreement dataAgreementForDataAttribute `json:"dataAgreement"`
+}
+
 type listDataAttributesResp struct {
 	DataAttributes interface{}         `json:"dataAttributes"`
 	Pagination     paginate.Pagination `json:"pagination"`
 }
 
-func dataAttributesToInterfaceSlice(dataAttributes []dataagreement.DataAttribute) []interface{} {
+func dataAttributesForList(res dataagreement.DataAgreement) []dataAttributeForLists {
```

Comment:
Why res ? change this to da (since you are trying to avoid collision)

Reply:
@georgepadayatti Done